src/game/phase/build.cpp

```diff
@@ -10,6 +10,21 @@ void BuildPhase::s_setup() {
         context.ready_flags[player_id] = true;
     });
 
+    s_check_funds_conn = events::build::s_check_funds_event.connect([&](proto::Construct& c) {
```

Comment:
A bit of a misnomer, since the event handler also has a "build this if you can afford it" side-effect. A name change and increased documentation is advised.